src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs

```diff
@@ -104,6 +108,17 @@ public void MarkConnectionAsNotIdle()
             _connectionMetrics?.IdleStateChanged(idle: false);
         }
 
+        public void LinkRequestActivity()
+        {
+            Activity? requestActivity = Activity.Current;
```

Comment:
Nit: can we check `_activity is null` before looking up `Activity.Current` to make it a bit cheaper when disabled?